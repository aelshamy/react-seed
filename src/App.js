import React, { Component, Fragment } from "react";

class App extends Component {
  render() {
    return (
      <Fragment>
        <header>
          <h1>React App</h1>
        </header>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Temporibus
          repudiandae distinctio optio culpa, ex debitis est quisquam, dolores
          explicabo unde ad? Esse corporis odit rerum itaque totam ad inventore
          voluptate!
        </p>
      </Fragment>
    );
  }
}

export default App;
