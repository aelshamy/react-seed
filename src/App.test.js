import React from "react";
import { create } from "react-test-renderer";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import App from "./App";

describe("<App/>", () => {
  beforeEach(() => {});

  test("it will match the snapshot", () => {
    const c = create(<App />);
    expect(toJson(c)).toMatchSnapshot();
  });

  describe("render()", () => {
    test("should render correctly", () => {
      const component = shallow(<App />);
      expect(component).toMatchSnapshot();
      expect(component.find("header").exists()).toBe(true);
      expect(component.find("header").text()).toEqual("React App");
      expect(component.find("p").exists()).toBe(true);
    });
  });
});
