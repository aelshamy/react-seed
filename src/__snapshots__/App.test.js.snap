// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> it will match the snapshot 1`] = `undefined`;

exports[`<App/> render() should render correctly 1`] = `
<Fragment>
  <header>
    <h1>
      React App
    </h1>
  </header>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Temporibus repudiandae distinctio optio culpa, ex debitis est quisquam, dolores explicabo unde ad? Esse corporis odit rerum itaque totam ad inventore voluptate!
  </p>
</Fragment>
`;
